feat(core/execution-parameters): condense parameters/artifacts and make it collapsable

kevinawoo opened the pull request:
This PR address some issues when there's lots of parameters and/or artifacts, making the pipeline page very long. On smaller screen sizes, unusable.
Included are some screenshots of what it looks like now and how it looks after applying this PR:

<details><summary>How it currently looks</summary>

<kbd>![](https://cl.ly/cb264f2abc58/Screen%20Shot%202019-03-26%20at%2010.12.30.png)</kbd>

</details>


<details open="true"><summary>Let's see a gif in action (download is kind of big, give it a sec)</summary>

<kbd>![ezgif com-gif-maker (3)](https://user-images.githubusercontent.com/3469532/55020621-6c3aa100-4fb4-11e9-8e0f-652123035df4.gif)</kbd>

</details>



<details><summary>Collapsed View with some annotations of what's changed</summary>

<kbd>![](https://cl.ly/1afe915e11e9/Image%202019-03-26%20at%2010.07.37.png)</kbd>

</details>




<details><summary>Expanded View with some annotations of what's changed</summary>

<kbd>![](https://cl.ly/2fba43ad0b58/Image%202019-03-26%20at%2010.28.56.png)</kbd>

</details>


<details><summary>Standalone View is auto expanded</summary>

<kbd>![](https://cl.ly/44ef50a83bb0/Screen%20Shot%202019-03-26%20at%2011.24.54.png)</kbd>

</details>

maggieneterval committed:
feat(core): add markdown functionality to custom app banners (#6745)

clareliguori committed:
fix(ecs): Fix name of health check grace period attribute (#6746)

Fixes spinnaker/spinnaker#4123

kevinawoo commented on app/scripts/modules/core/src/pipeline/executions/executions.less:
```diff
@@ -84,7 +84,7 @@
       &.group-by-name {
         .execution-bar {
-          padding-top: 20px;
+          //padding-top: 20px;
```
need to look at this

kevinawoo replied:
does it look weird?

alperkokmen committed:
fix(trigger): add webhook to triggerTypes (#6747)

webhook was missing from triggerTypes in settings.js whereas
it was available in halfconfig/settings.js. this brings parity between
these two configuration files.

anotherchrisberry committed:
feat(amazon): show certificate upload/expiration when selected for load balancers (#6731)

christopherthielen committed:
fix(core/projects): Submit task against the 'spinnaker' application (#6748)

kevinawoo commented on app/scripts/modules/core/src/pipeline/status/artifactList.less:
```diff
@@ -8,3 +10,43 @@
+    font-size: 0.8em;
+    line-height: 12px;
+    margin-bottom: 10px;
+    padding-left: @execution-details-indent - 0.2em;
```
add comment on why there's a 0.2em

kevinawoo commented on app/scripts/modules/core/src/pipeline/status/executionParameters.less:
```diff
@@ -0,0 +1,39 @@
+    line-height: 12px;
+    overflow: scroll;
+    margin-bottom: 10px;
+    padding-left: @execution-details-indent - 0.2em;
```
add comment on why there's a 0.2em

Commit:
fix(google): add better help text around accelerators (#6750)

Commit:
fix(IAP) Fix refresher not enabling (#6740)

* fix(IAP): Resolve dodgy syntax in template file

* fix(core): Grey out execution actions for mptv2 pipelines (#6734)

https://github.com/spinnaker/spinnaker/issues/4167

* fix(concourse): Support manual trigger execution by build number (#6738)

* feat(mptv2): add a search input to the mptv2 list (#6742)

Commit:
fix(google): select all zones by default when deploying a regional gce server group (#6751)

ncknt commented on app/scripts/modules/core/src/pipeline/executions/execution/Execution.tsx:
```diff
@@ -379,7 +415,45 @@ export class Execution extends React.Component<IExecutionProps, IExecutionState>
+              artifacts={artifacts}
+              resolvedExpectedArtifacts={resolvedExpectedArtifacts}
+              showingExpandedArtifacts={showingParams}
+              columnFormatAfter={Math.min(collapseParamsArtifactsAfter, collapseParamsArtifactsAfter - parametersCount)}
```
Remove the Math.min() -> `collapseParamsArtifactsAfter - parametersCount`

anotherchrisberry committed:
refactor(titus): generalize ConfigBin and export it (#6757)

* refactor(titus): export configBin components, make more configurable

* refactor(amazon): make ScalingPolicyDetailsSection overridable

anotherchrisberry committed:
Bump package core to 0.0.346 and amazon to 0.0.182 and titus to 0.0.81 (#6758)

* chore(core): Bump version to 0.0.346

4bc6ecd3ee0a3938627cb8b6c8a9c3c68148b3f8 fix(core/projects): Submit task against the 'spinnaker' application (#6748)
1f58407a8f8d1c28bf14c0e372cd3b482f4bb526 feat(amazon): show certificate upload/expiration when selected for load balancers (#6731)
efcec9696f70153ea09c4b0763ee15466a40c563 feat(core): add markdown functionality to custom app banners (#6745)
9c14b02e3212bfce64d2e228273a5ae2a2907fb6 feat(mptv2): add a search input to the mptv2 list (#6742)
a84eda8efb966221756c4ccad7a5106722892c6b fix(concourse): Support manual trigger execution by build number (#6738)
5529d53d562e9a8a6884ef93f96296c3d703b998 fix(core): Grey out execution actions for mptv2 pipelines (#6734)
1c9e07543aaefd36d8e1a5ea9d51321c350b57a8 feat(ecs): docker image selection (#6687)
2fd608a1eb238fb352fbccedbce506195801f02d feat(mptv2): add list screen for pipeline templates (#6723)
5cd0aa0c263247153cb809929782ffff6faed698 fix:(core): Replace copy icon with descriptive text in template modal (#6725)
0d68c809a4b1ebbaa32204691de00638dd2133f2 refactor(forms): Creating contexts for Layout and Help (#6718)
7ab32f98df1ef12957002372937b13485e3fd9a2 fix(stages): Do not delete stageTimeoutMs when rendering (#6729)
03fa0d962d743318bc63354676b2ea15c2216831 fix(artifacts): Exclude unmatchable expected artifact types (#6709)

* chore(amazon): Bump version to 0.0.182

f32aa23030642d8e403d731038eb60c97915c1d9 refactor(titus): generalize ConfigBin and export it (#6757)
1f58407a8f8d1c28bf14c0e372cd3b482f4bb526 feat(amazon): show certificate upload/expiration when selected for load balancers (#6731)
6083b3f4ea9b7fc66c2666a929bde70c35ef2275 fix(amazon): consider all target group names when creating new ones (#6727)

* chore(titus): Bump version to 0.0.81

f32aa23030642d8e403d731038eb60c97915c1d9 refactor(titus): generalize ConfigBin and export it (#6757)

anotherchrisberry committed:
refactor(titus): export more config bin stuff (#6759)

anotherchrisberry committed:
chore(titus): Bump version to 0.0.82 (#6760)

fb6eeb4f2b4c10fc6aeabba25eaa4ffb21ceece0 refactor(titus): export more config bin stuff (#6759)

anotherchrisberry committed:
style(titus): tweak styles for config bin modal (#6762)

anotherchrisberry committed:
chore(titus): Bump version to 0.0.83 (#6763)

5474d6278bb4762c73b138ff567ddbcb3df15ca5 style(titus): tweak styles for config bin modal (#6762)

anotherchrisberry committed:
fix(core): send application when paging via PagerDuty (#6764)

anotherchrisberry committed:
chore(core): Bump version to 0.0.347 (#6765)

65fbb679c66251db38e3db0ed62ce7f0c559ef1f fix(core): send application when paging via PagerDuty (#6764)

anotherchrisberry committed:
refactor(titus): remove config bin code (#6767)

anotherchrisberry committed:
fix(titus): remove metricOptions field (#6768)

anotherchrisberry committed:
chore(titus): Bump version to 0.0.84 (#6769)

e6a305e7c922f481f9cd14ddce6c14b707659ed3 fix(titus): remove metricOptions field (#6768)
c64a876822532cd791d5672442df1fc419073a85 refactor(titus): remove config bin code (#6767)

anotherchrisberry committed:
fix(titus): send cloudProvider as a string to ConfigBinLink (#6770)

anotherchrisberry committed:
chore(titus): Bump version to 0.0.85 (#6771)

4b8e48dd65dc60e0cfbd11a2ee7567b6d58302e6 fix(titus): send cloudProvider as a string to ConfigBinLink (#6770)

maggieneterval committed:
fix(core): make chaos monkey opt-in for new apps (#6766)

anotherchrisberry committed:
feat(core): provide link when task/stage fails due to traffic guards (#6772)

anotherchrisberry committed:
chore(core): Bump version to 0.0.348 (#6773)

ad8d504863f654a659dc7c82b97b1c1b7a61a7e0 feat(core): provide link when task/stage fails due to traffic guards (#6772)
d819cf3740f9341f66aee9c8cfa40eb01c404054 fix(core): make chaos monkey opt-in for new apps (#6766)

maggieneterval committed:
feat(gcb): add Google Cloud Build stage (#6774)

* refactor(core): move yamlEditor from kubernetes to core module

* feat(gcb): add Google Cloud Build stage

maggieneterval committed:
fix(kubernetes): fix validation for Find Artifact from Resource stage (#6777)

maggieneterval committed:
fix(kubernetes): safe lookups for apiVersion on patch manifest deploy status (#6775)

Commit:
feat(templates): add Save button to export pipeline json modal (#6761)

Commit:
fix(lint): Fix lint in pipeline templates list (#6782)

jkschneider committed:
fix(artifacts): Persist default artifact account in ExpectedArtifactModal (#6783)

erikmunson committed:
chore(deps): bump @spinnaker/kayenta from 0.0.73 to 0.0.74 (#6786)

jkschneider committed:
fix(pipeline): Fix target impedance validator for clone server group (#6785)

mattrjacobs committed:
fix(gremlin): Changed flag from features.gremlinEnabled to features.gremlin (#6776)

erikmunson committed:
fix(core/pagerDuty): give appropriate app to PagerDutyWriter (#6789)

erikmunson committed:
chore(core): Bump version to 0.0.349 (#6790)

8b1e5bf48a9d07007408d3eed9e355f6180857fb fix(core/pagerDuty): give appropriate app to PagerDutyWriter (#6789)
8422d6a2312ff08577fdcdbc74c49e107b2b06b2 fix(pipeline): Fix target impedance validator for clone server group (#6785)
ce12338f0d90925321a71299099bad162480fce8 fix(artifacts): Persist default artifact account in ExpectedArtifactModal (#6783)
3519f6f9e7d479c19755e2e40f46f323b0a82cf3 fix(lint): Fix lint in pipeline templates list (#6782)
36b86da691b44174a69984b06036c40e4900ad49 feat(templates): add Save button to export pipeline json modal (#6761)
50b74f2d0db86b15b66fd7a9a4985d0efb8fdb4b feat(gcb): add Google Cloud Build stage (#6774)

louisjimenez committed:
feat(core): Create pipeline from v2 template list

https://github.com/spinnaker/spinnaker/issues/3893

anotherchrisberry committed:
fix(amazon): allow deletion of scaling policies without alarms (#6779)

doubret committed:
fix(amazon): Added AWS/ApplicationELB namespace to cloudwatch namespaces (#6791)

clareliguori committed:
fix(ecs): populate load balancers when configuring cmd (#6778)

anotherchrisberry committed:
Bump package amazon to 0.0.183 and titus to 0.0.86 (#6793)

* chore(amazon): Bump version to 0.0.183

607f386a3511da4bb568a672ea2447ca2038b622 fix(amazon): Added AWS/ApplicationELB namespace to cloudwatch namespaces (#6791)
e96e7a927c56540c998fd6d09134df8438978625 fix(amazon): allow deletion of scaling policies without alarms (#6779)

* chore(titus): Bump version to 0.0.86

e96e7a927c56540c998fd6d09134df8438978625 fix(amazon): allow deletion of scaling policies without alarms (#6779)

Commit:
feat(mptv2): Add delete button & confirm modal to pipeline templates list (#6792)

jkschneider committed:
fix(provider/cf): Make expected artifacts selectable as clone manifests (#6796)

jkschneider committed:
fix(triggers): Add lastSuccessfulBuild as a build option in Jenkins default artifact (#6797)

maggieneterval committed:
fix(kubernetes): show Deployment clusters in Find Artifacts from Resource stages (#6794)

maggieneterval committed:
fix(google): fix autohealing health checks in deploy stages (#6804)

anotherchrisberry committed:
refactor(core): de-angularize ApplicationModelBuilder, fix project executions (#6802)

anotherchrisberry committed:
fix(amazon): hide security groups on NLBs (#6803)

anotherchrisberry committed:
Bump package core to 0.0.350 and amazon to 0.0.184 (#6807)

* chore(core): Bump version to 0.0.350

72e164dfcf11afbac559195a8fa6a91dd77ad2b9 refactor(core): de-angularize ApplicationModelBuilder, fix project executions (#6802)
cc053391ad7d11725066f8b25dcbe1853e4120cd fix(triggers): Add lastSuccessfulBuild as a build option in Jenkins default artifact (#6797)
daf72c4c0076af68d4d85720b3efc99454ba2f8f feat(mptv2): Add delete button & confirm modal to pipeline templates list (#6792)

* chore(amazon): Bump version to 0.0.184

06a7556ca5c3e7d78a4fc629b90854fae391f193 fix(amazon): hide security groups on NLBs (#6803)
72e164dfcf11afbac559195a8fa6a91dd77ad2b9 refactor(core): de-angularize ApplicationModelBuilder, fix project executions (#6802)

maggieneterval committed:
fix(google): GCE create server group and load balancer fixes (#6806)

* fix(google): safe call of onTypeChanged in v2InstanceTypeSelector

* fix(google): safe lookup and set of namedPorts in loadBalancingPolicySelector

Commit:
fix(google): revert "select all zones by default when deploying a regional gce server group (#6751)" (#6808)

This reverts commit ad2e2227442aa2e7abf85bf1a82ef7a5f51b1445.

In hindsight, Deck was already doing the right thing here. GCP chooses
the zones and, in some quick testing, the MIGs in us-central1 always
seem to get spread across zones -b, -c, and -f.

Commit:
fix(artifacts): default helm artifact editor is broken (#6811)

jasonjchu committed:
fix(provider/cf): Fix provider selection for resize stage in pipeline (#6754)

Commit:
fix(artifacts): helm artifact, replace object assignw with object spread (#6814)

Commit:
fix(mptv2): clicking a view link on template list screen throws exception (#6815)

Commit:
feat(provider/azure): Add redblack strategy for azure (#6801)

dibyom committed:
fix(triggers): Remove RunAsUser if pipeline permissions enabled for react triggers. (#6818)

maggieneterval committed:
chore(kubernetes): refactor BasicSettings component to be usable in stages (#6820)

jkschneider committed:
fix(provider/cf): Clone model so Orca monitors the correct foundation for up instances (#6817)

louisjimenez committed:
Merge pull request #6819 from louisjimenez/create-pipeline-from-template

feat(core): Create pipeline from v2 template list

Commit:
feat(cf): Create Service Key pipeline stage (#6821)

spinnaker/spinnaker#4242

Co-Authored-By: Ria Stein <eleftheria.kousathana@gmail.com>

maggieneterval committed:
feat(kubernetes): feature-flagged support for kubernetes traffic management strategies (#6816)

kevinawoo committed:
feat(core/execution-parameters): condense parameters/artifacts and make it collapsable

kevinawoo committed:
chore(github): add some tips to improve pull requests (#6825)

* chore(github): add tips to improve pull requests

* english is hard

Commit:
feat(cf): add deployment strategy to clone SG (#6827)

christopherthielen committed:
Merge branch 'master' into feature/display-lots-of-parameters
